### PR TITLE
wire nanotabicl regression

### PR DESCRIPTION
**Pull request by borawhocodess**
in this pr:
- made `dist` optional on `TabularRegressor`
    - averages the quantiles for nanotabicl point estimate
- added `QuantileLoss` to `utils.py`
    - pinball loss summed over the 999 quantiles
    - matches the tabiclv2 paper
    - nanotabicl ships no loss
    - [tabicl repo](https://github.com/soda-inria/tabicl/blob/8f665edb1e3099907e15bf645bb319fbb2f99c24/src/tabicl/_finetune/regressor.py#L19) only has finetuning
- made `train.py` accept `QuantileLoss` as a regression criterion
- initialised config files with paper defaults:
    - `nanotabicl_classifier_pretraining.yaml`
    - `nanotabicl_regressor_pretraining.yaml`
    - added `_pretraining` extension
        - because there is also training configs included
        - can be changed later

not here: (tracked in my todos)
- full tabicl decode (tail-corrected mean, monotonicity, ensemble, median/variance)
- pretraining integration

### CODE REVIEW

**Comment by borawhocodess** [comment]
i lost my todos :'( during my computer reset

File: tfmplayground/interface.py [hunk 1]
```diff
@@ -159,7 +157,7 @@ def predict(self, X_test: np.ndarray) -> np.ndarray:
             y_train = torch.from_numpy(self.y_train_n).unsqueeze(0).to(torch.float).to(self.device)
 
             logits = self.model(X_train, y_train, X_test).squeeze(0)
-            preds_n = self.dist.mean(logits)
+            preds_n = self.dist.mean(logits) if self.dist is not None else logits.mean(dim=-1)
```
Comment (AlexanderPfefferle) [inline tfmplayground/interface.py:160]: Merged, but I am a bit sus on this

Comment (borawhocodess) [reply tfmplayground/interface.py:160]: some late supportive findigs:

in [tabiclv2 paper](https://arxiv.org/html/2602.11139v1#:~:text=At%20inference%2C%20for%20point%20estimation%20we%20simply%20average%20the%20predicted%20quantiles%2C%20which%20proves%20both%20fast%20and%20effective.):

> "At inference, for point estimation we simply average the predicted quantiles, which proves both fast and effective."

confirmed in tabicl inference code at [src/tabicl/_model/tabicl.py#L594](https://github.com/soda-inria/tabicl/blob/54abea15b3379ab12b2924bc7ffc89ea1102d9d4/src/tabicl/_model/tabicl.py#L594)

```
    output_type: str = "mean"
    ...
    if "mean" in output_type:
        results["mean"] = raw_quantiles.mean(dim=-1)
```

also our `logits` is not really logits here, variable name is stuck from the earlier classification code, quantiles sit behind the var name `logits`
